integration/service/test/geopm_report_cron_plot.py

```diff
@@ -0,0 +1,242 @@
+#  Copyright (c) 2015 - 2024 Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+import yaml
+import sys
+import os
+from glob import glob
+from pandas import DataFrame, to_datetime, to_timedelta
+import plotly.graph_objs as go
+from dash import Dash, dcc, html
+from argparse import ArgumentParser
+
+class CronReport:
+    def __init__(self, reports='geopm-report-*.yml'):
+        if type(reports) is str:
+            report_names = glob(reports)
+        else:
+            report_names = reports
+        report_list = []
+        for rn in report_names:
+            with open(rn) as fid:
+                report = yaml.load(fid, Loader=yaml.CSafeLoader)
+            if report is None:
+                sys.stderr.write(f'Warning: {rn} is empty, could not be parsed\n')
+                continue
+            df_row_keys = set(report) - {'metrics'}
+            df_row = dict()
+            for df_key in df_row_keys:
+                if df_key == 'sample-time-first':
+                    df_row[df_key] = to_datetime(report[df_key])
+                else:
+                    df_row[df_key] = report[df_key]
+            for metric_name, metrics in report['metrics'].items():
+                df_row['metric-name'] = metric_name
+                for m_key, m_value in metrics.items():
+                    df_row[m_key] = m_value
+                report_list.append(df_row.copy())
+        self._df = DataFrame(report_list)
+        self._df.sort_values('sample-time-first', inplace=True)
+        if len(self._df) == 0:
+            raise RuntimeError('Could not load reports')
+        self._colors = dict(
+            mean_marker='rgb(247, 240, 25)',
+            min_marker='rgb(82, 115, 222)',
+            max_marker='rgb(217, 85, 85)',
+            std_marker='rgb(68, 68, 68)',
+            std_fill='rgba(68, 68, 68, 0.5)',
+            range_fill='rgba(68, 68, 68, 0.3)',
+            pie_line='rgb(0, 0, 0)',
+            pie_gpu='red',
+            pie_cpu='orange',
+            pie_dram='yellow',
+            background='rgb(41, 128, 185)',
+            page_text='rgb(252, 252, 252)',
+        )
+        self._font_family = 'Lato, proxima-nova, Helvetica Neue, Arial, sans-serif'
+
+
+    def calculate_energy(self, domain, begin_date, end_date):
+        domain = domain.upper()
+        metric_name = f'{domain}_POWER'
+        time = self.get_stat(begin_date, end_date, metric_name, 'sample-time-first')
+        time_total = self.get_stat(begin_date, end_date, metric_name, 'sample-time-total')
+        power = self.get_stat(begin_date, end_date, metric_name, 'mean-arithmetic')
+        kwh_factor = 1.0 / 3.6e6
+        # TODO include energy from last report as well
+        return kwh_factor * sum((power.shift(1) * (time - time.shift(1)).dt.total_seconds()).dropna())
```

Comment:
Maybe add a comment explaining why it's insufficient to simply sum each report's `T*P`.